File: 2855-maximum-number-of-jumps-to-reach-the-last-index/maximum-number-of-jumps-to-reach-the-last-index.py
class Solution:
    def maximumJumps(self, nums: List[int], t: int) -> int:
        n=len(nums)

        dp=[float("-inf")]*n
        dp[0]=0
        for i in range(1, n):
            for j in range(0, i):
                if -t<=nums[i]-nums[j] and nums[i]-nums[j]<=t and 1+dp[j]>dp[i]:
                    if(dp[j]>-1):
                        dp[i]=max(dp[i], 1+dp[j])
                    
        return -1 if dp[n-1] == float("-inf") else dp[n-1]

File: Medium/2855-maximum-number-of-jumps-to-reach-the-last-index/maximum-number-of-jumps-to-reach-the-last-index.cpp
class Solution{
public:
    int maximumJumps(vector<int> &nums, int t){
        int n=nums.size();

        vector<int> dp(n, INT_MIN);
        dp[0]=0;
        for(int i=1; i<n; i++){
            for(int j=0; j<i; j++){
                if(-t<=nums[i]-nums[j] && nums[i]-nums[j]<=t && 1+dp[j]>dp[i]){
                    if(dp[j]>-1){
                        dp[i]=max(dp[i], 1+dp[j]);
                    }
                }
            }
        }

        if(dp[n-1] == INT_MIN)
            return -1;

        return dp[n-1];
    }
};